fog/ledger/server/tests/router_connection.rs

```diff
@@ -0,0 +1,908 @@
+            "insecure-mca://127.0.0.1:{}",
+            pick_unused_port().expect("No free ports")
+        ))
+        .unwrap();
+        let config = LedgerRouterConfig {
+            chain_id: "local".to_string(),
+            ledger_db: db_full_path.to_path_buf(),
+            watcher_db: watcher_dir,
+            admin_listen_uri,
+            client_listen_uri: client_listen_uri.clone(),
+            client_responder_id: ResponderId::from_str(&client_listen_uri.addr()).unwrap(),
+            ias_spid: Default::default(),
+            ias_api_key: Default::default(),
+            client_auth_token_secret: None,
+            client_auth_token_max_lifetime: Default::default(),
+            query_retries: 3,
+            omap_capacity: OMAP_CAPACITY,
+        };
+
+        let enclave = LedgerSgxEnclave::new(
+            get_enclave_path(mc_fog_ledger_enclave::ENCLAVE_FILE),
+            &config.client_responder_id,
+            OMAP_CAPACITY,
+            logger.clone(),
+        );
+
+        let grpc_env = Arc::new(grpcio::EnvBuilder::new().build());
+
+        let ra_client =
+            AttestClient::new(&config.ias_api_key).expect("Could not create IAS client");
+        let mut ledger_server = LedgerRouterServer::new(
+            config,
+            enclave,
+            ra_client,
+            Arc::new(RwLock::new(HashMap::new())),
+            ledger.clone(),
+            watcher.clone(),
+            logger.clone(),
+        );
+
+        ledger_server.start();
+
+        // Make unattested ledger client
+        let client = FogUntrustedLedgerGrpcClient::new(
+            client_listen_uri,
+            GRPC_RETRY_CONFIG,
+            grpc_env,
+            logger.clone(),
+        );
+
+        // Get a tx_out that is actually in the ledger
+        let real_tx_out0 = { ledger.get_tx_out_by_index(0).unwrap() };
+
+        // Try to get tx out records
+        let queries: Vec<CompressedRistrettoPublic> =
+            vec![(&[0u8; 32]).into(), real_tx_out0.public_key];
+        let result = client.get_tx_outs(queries).unwrap();
+        // Check that we got expected num_blocks value
+        assert_eq!(result.num_blocks, 4);
+        // Check that we got 2 results, as expected
+        assert_eq!(result.results.len(), 2);
+        assert_eq!(
+            &result.results[0].tx_out_pubkey.clone().unwrap().data[..],
+            &[0u8; 32]
+        );
+        assert_eq!(result.results[0].result_code, TxOutResultCode::NotFound);
+        assert_eq!(
+            &result.results[1].tx_out_pubkey.clone().unwrap().data[..],
+            &real_tx_out0.public_key.as_bytes()[..]
+        );
+        assert_eq!(result.results[1].result_code, TxOutResultCode::Found);
+        assert_eq!(result.results[1].tx_out_global_index, 0);
+        assert_eq!(result.results[1].block_index, 0);
+        assert_eq!(
+            result.results[1].timestamp_result_code,
+            TimestampResultCode::BlockIndexOutOfBounds as u32
```

Comment:
:question: This may get answered in the previous test, but why is this out of bounds?